### PR TITLE
Add github action to check flake groups

### LINKED ISSUE #356: Validate flakes/*.toml format
PR's changing the flake lists should be in the correct format

Would a `--dry-run` flag be an option?

### LINKED ISSUE #430: Add a github action to test the integrity of `manual.toml`

### PR DESCRIPTION
**Pull request by ysndr**
Resolves #430 (and closes #356)

### CODE REVIEW

File: .github/workflows/check-flake-files.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+      with:
+        name: nixos-search
+        signingKey: '${{ secrets.CACHIX_SIGNING_KEY }}'
+
+    - name: Install unstable channel
+      run: |
+        nix-channel --add https://nixos.org/channels/nixpkgs-unstable
+        nix-channel --update
+
+
```
Comment (turboMaCk) [inline .github/workflows/check-flake-files.yml:46]:
```suggestion
```

File: .github/workflows/check-flake-files.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        nix-channel --update
+
+   
+    - name: Building import_scripts
+      run: |
+        nix build ./#packages.x86_64-linux.flake-info
+
+    - name: Try importing all custom flakes
+      run: |
+
```
Comment (turboMaCk) [inline .github/workflows/check-flake-files.yml:53]:
```suggestion
```

**Comment by turboMaCk** [comment]
I'm not that experienced with flakes to be able to review this properly. Anyway CI pass is a good sign.

**Review by turboMaCk — APPROVED** [review]

File: .github/workflows/check-flake-files.yml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    env:
+      RUST_LOG: debug
+    
+    steps:
+
+    - name: Checking out the repository
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
```
Comment (ncfavier) [inline .github/workflows/check-flake-files.yml:25]: Why? This fetches the entire history.

File: .github/workflows/check-flake-files.yml [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+    - name: Installing Nix
+      uses: cachix/install-nix-action@v16
+      with:
+        install_url: https://nixos-nix-install-tests.cachix.org/serve/i6laym9jw3wg9mw6ncyrk6gjx4l34vvx/install
+        install_options: '--tarball-url-prefix https://nixos-nix-install-tests.cachix.org/serve'
+        extra_nix_config: |
+          experimental-features = nix-command flakes
+          access-tokens = github.com=${{ secrets.GITHUB_TOKEN }}
```
Comment (ncfavier) [inline .github/workflows/check-flake-files.yml:34]: Do we need any of this? Latest nix should be fine, and flakes are enabled by default

File: .github/workflows/check-flake-files.yml [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+    - uses: cachix/cachix-action@v10
+      with:
+        name: nixos-search
+        signingKey: '${{ secrets.CACHIX_SIGNING_KEY }}'
+
+    - name: Install unstable channel
+      run: |
+        nix-channel --add https://nixos.org/channels/nixpkgs-unstable
+        nix-channel --update
```
Comment (ncfavier) [inline .github/workflows/check-flake-files.yml:44]: Do we need this?

File: .github/workflows/check-flake-files.yml [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+    - name: Building import_scripts
+      run: |
+        nix build ./#packages.x86_64-linux.flake-info
```
Comment (ncfavier) [inline .github/workflows/check-flake-files.yml:48]:
```suggestion
        nix build .#flake-info
```

File: .github/workflows/check-flake-files.yml [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+
+    - name: Try importing all custom flakes
+      run: |
+        for flake_group in $(find ./flakes -name "*.toml")
+        do
+          ./result/bin/flake-info group $flake_group $(basename $flake_group)
```
Comment (ncfavier) [inline .github/workflows/check-flake-files.yml:54]:
```suggestion
        for flake_group in flakes/**/*.toml
        do
          ./result/bin/flake-info group "$flake_group" "$(basename "$flake_group" .toml)"
```

**Review by ncfavier — CHANGES_REQUESTED** [review]

File: .github/workflows/check-flake-files.yml (hunk 4 above)
Comment (ncfavier) [reply .github/workflows/check-flake-files.yml:34]: Never mind, fixed in #433

File: .github/workflows/check-flake-files.yml (hunk 5 above)
Comment (ncfavier) [reply .github/workflows/check-flake-files.yml:44]: Never mind, fixed in #433

File: .github/workflows/check-flake-files.yml (hunk 3 above)
Comment (ysndr) [reply .github/workflows/check-flake-files.yml:25]: I don't know, was there before just moved it.

Comment (ysndr) [reply .github/workflows/check-flake-files.yml:25]: Wait, wrong PR the moving happens in the next one, sorry for the spoiler
Anyway, I don't know why, will fix it in #433

**Review by garbas — APPROVED** [review]

**Pull request merged**